src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -1539,18 +1549,25 @@ protected long getNextDelayInMillis() {
     protected long getDelayEstimateInMs() {
       return nextRunTime - Clock.lastKnownForwardProgressingMillis();
     }
-    
+
     @Override
-    public void executing() {
+    public boolean canExecute() {
       if (canceled) {
-        return;
+        return false;
+      }
+      synchronized (queueSet.scheduleQueue.getModificationLock()) {
```

Comment:
I feel like this should be doable w/o the lock. assuming the remove is atomic and would only return true for the call that actually removes.